components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -9238,6 +9243,19 @@ public void addKeyManagerConfiguration(KeyManagerConfigurationDTO keyManagerConf
                 preparedStatement.setString(9, keyManagerConfigurationDTO.getTokenType());
                 preparedStatement.setString(10, keyManagerConfigurationDTO.getExternalReferenceId());
                 preparedStatement.executeUpdate();
+                KeyManagerPermissionConfigurationDTO permissionDTO = keyManagerConfigurationDTO.getPermissions();
+                if (permissionDTO != null && permissionDTO.getPermissionType() != "PUBLIC") {
```

Comment:
Shall we use a constant for "PUBLIC"? I saw a couple of places that use this string.

Reply:
Fixed in f3c1bd0509cf7ce03759bdb6c142ada52acb33c2